### PR TITLE
add support one or more property path (+)

### PR DESCRIPTION
**Pull request by patofp**
# Description

This PR adds support for one or more length property path. E.g:

```
PREFIX foaf: <http://xmlns.org/foaf/0.1/>

SELECT ?s ?o
WHERE {
 ?s foaf:knows+ ?o .
}
```

## Type of change

Make sure you add the correct label for the PR:

- `enhacement` if this PR adds a new feature
- `bug` if it fixes a bug
- `documentation` if it adds docs
- `breaking-change` if this PR introduces a breaking change
- `dependency-updates` if it updates dependencies

### CODE REVIEW

File: modules/engine/src/main/scala/com/gsk/kg/engine/properties/FuncProperty.scala [hunk 1]
```diff
@@ -91,6 +91,157 @@ object FuncProperty {
...
+
+    implicit val spoEncoder: Encoder[Row] = RowEncoder(
+      StructType(
+        List(
+          StructField("s", StringType),
+          StructField("p", StringType),
+          StructField("o", StringType)
+        )
+      )
+    )
```
Comment (pepegar) [inline modules/engine/src/main/scala/com/gsk/kg/engine/properties/FuncProperty.scala:226]: Don't we have this very same encoder somewhere else?  I think we should extract it to a common place

File: modules/engine/src/main/scala/com/gsk/kg/engine/properties/FuncProperty.scala [hunk 2]
```diff
@@ -91,6 +91,157 @@ object FuncProperty {
...
+    * more triples to connect.
+    * @param baseDf Base dataframe with all the triples unconnected
+    * @param accDf The dataframe which it will accumulate every depth level of paths
+    * @param i Iteration index
+    * @param sIdx Subject index (for column renaming)
+    * @param pIdx Predicate index (for column renaming)
+    * @param oIdx Object index (for column renaming
+    * @return Dataframe with all the triples connected creating paths
+    */
+  private def traverse(
```
Comment (pepegar) [inline modules/engine/src/main/scala/com/gsk/kg/engine/properties/FuncProperty.scala:138]: What do you think about changing the name to something more specific?

**Review by pepegar — APPROVED** [review]
Added a  couple of cosmetic comments, but this looks amazing!

File: modules/engine/src/main/scala/com/gsk/kg/engine/properties/FuncProperty.scala (hunk 1 above)
Comment (patofp) [reply modules/engine/src/main/scala/com/gsk/kg/engine/properties/FuncProperty.scala:226]: Yeah, good idea!

**Pull request merged**